frontend/packages/erd-core/src/features/erd/components/ERDContent/hooks/usePopStateListener.ts

```diff
@@ -9,35 +12,69 @@ import {
   getHiddenNodeIdsFromUrl,
   getShowModeFromUrl,
 } from '@/utils'
-import { useEffect } from 'react'
+import type { Node } from '@xyflow/react'
+import { useCallback, useEffect } from 'react'
+import { hasNonRelatedChildNodes, updateNodesHiddenState } from '../utils'
 
-export const usePopStateListener = () => {
-  useEffect(() => {
-    const handlePopState = async () => {
-      const tableName = getActiveTableNameFromUrl()
-      const showMode = getShowModeFromUrl()
-      const hiddenNodeIds = await getHiddenNodeIdsFromUrl()
+type Params = {
+  nodes: Node[]
+  displayArea: DisplayArea
+}
+
+export const usePopStateListener = ({ nodes, displayArea }: Params) => {
+  const { getEdges, setNodes, setEdges, fitView } = useCustomReactflow()
 
-      updateIsPopstateInProgress(true)
+  const handlePopState = useCallback(async () => {
+    updateIsPopstateInProgress(true)
 
-      await new Promise<void>((resolve) => {
+    const [tableName, showMode, hiddenNodeIds] = await Promise.all([
+      getActiveTableNameFromUrl(),
+      getShowModeFromUrl(),
+      getHiddenNodeIdsFromUrl(),
+    ])
```

Comment:
[nits]
Since there's only one asynchronous function, I thought it's fine to keep it as it was before using `Promise.all`.

```suggestion
    const tableName = getActiveTableNameFromUrl()
    const showMode = getShowModeFromUrl()
    const hiddenNodeIds = await getHiddenNodeIdsFromUrl()
```

Reply:
@junkisai i have commited this change